virtualizarr/xarray.py

```diff
@@ -322,7 +321,7 @@ def separate_coords(
             # use workaround to avoid creating IndexVariables described here https://github.com/pydata/xarray/pull/8107#discussion_r1311214263
             if len(var.dims) == 1:
                 dim1d, *_ = var.dims
-                coord_vars[name] = (dim1d, var.data)
+                coord_vars[name] = (dim1d, var.data, var.attrs)
```

Comment:
This gets passed into the constructor of [`xr.Coordinates`](https://docs.xarray.dev/en/latest/generated/xarray.Coordinates.html), which calls [`xr.as_variable`](https://docs.xarray.dev/en/latest/generated/xarray.as_variable.html).